### PR TITLE
Fixed issue with yum package module regarding packages with epoch not validating

### PR DESCRIPTION
**Pull request by craigcomstock**

### CODE REVIEW

File: modules/packages/vendored/yum.mustache [hunk 1]
```diff
@@ -87,7 +87,7 @@ def get_package_data():
         # Absolute file.
         sys.stdout.write("PackageType=file\n")
         sys.stdout.flush()
-        return subprocess_call([rpm_cmd, "--qf", rpm_output_format, "-qp", pkg_string])
+        return subprocess_call([rpm_cmd, "--qf", rpm_output_format, "-qp", pkg_string, "|sed 's/(none)://'"])
```
Comment (craigcomstock) [inline modules/packages/vendored/yum.mustache:90]: will replace sed with python-native search/replace just to keep things tidy :)

**Review by nickanderson — COMMENTED** [review]
Makes sense to me but not a good person to judge the code

**Comment by craigcomstock** [comment]
> Makes sense to me but not a good person to judge the code

Thanks! I requested a few python coders review.

File: modules/packages/vendored/yum.mustache [hunk 2]
```diff
@@ -87,7 +87,16 @@ def get_package_data():
...
         sys.stdout.flush()
-        return subprocess_call([rpm_cmd, "--qf", rpm_output_format, "-qp", pkg_string])
+        process = subprocess_Popen([rpm_cmd, "--qf", rpm_output_format, "-qp", pkg_string], stdout=subprocess.PIPE)
+        (stdoutdata, _) = process.communicate()
+
+        if process.returncode != 0:
+            return process.returncode
+
+        for line in stdoutdata.decode("utf-8").splitlines():
+            sys.stdout.write(line.replace("(none):","") + "\n")
```
Comment (larsewi) [inline modules/packages/vendored/yum.mustache:97]:
```suggestion
            sys.stdout.write(line.replace("(none):", "") + "\n")
```

**Review by larsewi — APPROVED** [review]
`s/valgring/valgrind/g` in second commit. Looks good :rocket:

File: modules/packages/vendored/yum.mustache (hunk 2 above)
Comment (craigcomstock) [reply modules/packages/vendored/yum.mustache]: yeah, ran black on it as a separate commit now :+1:

**Comment by craigcomstock** [comment]
cherry picks
https://github.com/cfengine/masterfiles/pull/2972
https://github.com/cfengine/masterfiles/pull/2971

**Pull request merged**